Sources/Lifecycle/Lifecycle.swift

```diff
@@ -83,61 +83,66 @@ public struct LifecycleHandler {
 
 public struct ServiceLifecycle {
     private let configuration: Configuration
-    private let lifecycle: ComponentLifecycle
 
-    /// Creates a `ComponentLifecycle` instance.
+    /// The internal `ComponentLifecycle` instance
```

Comment:
```suggestion
    /// The underlying `ComponentLifecycle` instance
```

to avoid `internal` word which can be confusing?